### ISSUE TITLE
Override language for file ending in .j2

### ISSUE THREAD

**Issue by lonix1** [samuelcolvin/jinjahtml-vscode#83]
By default all files that end in `.j2` are matched to `jinja-html` (unless they have a more specific extension, e.g. `.sql.j2`).

The extension often incorrectly identifies files because of this. Linux config files usually don't have extensions (but they are templates, so end with `.j2`), so this extension always assumes `jinja-html`.

How can I override this setting, to `jinja-properties`? The problem is many of file extensions end with `.j2` so how do I specify just that one?

**Comment by samuelcolvin** [comment]
Not that I know of, but if you can find a way, I'd be happy to accept a PR.

You can always change the highlighting language using the dialog in the bottom right.

**Comment by lonix1** [comment]
Yeah that's what I do - click that language button - but it's a real chore to do that every time.

The only solutions I can think of:
- overriding `.j2` in my `settings.json` - but that would unset all the other file types as well... and there are dozens of them. :(
- being able to override the "default" mapping, e.g. the default would be `"jinjahtml.default": "jinja-html"`

Nonetheless, thanks for making this cool extension, it's quite useful!

**Comment by samuelcolvin** [comment]
I'm not entirely clear what you're asking for here?

Do you want to change the `.js` extension to highlight as `jinja-raw`?

This would be a a breaking change, but I think it might be reasonable. I wonder if anyone else has an opinion either way?

> I unfortunately don't have any skill at vscode extensions.

Me neither before I build this extension, luckily building extensions is quite well documented by Microsoft. Why don't you spend a bit of time researching settings for vscode extensions and see if you can fix this?

**Comment by lonix1** [comment]
That is why I made the other issue, as its a feature request which is self contained.

What I meant is, not to make a breaking change, but to make the default configurable and keep it as it is today: that `.j2` must open with `jinja-html`. So only those people who don't like that, can set it to something else, OR unset it (`""`) so it doesn't trigger the extension at all.

Smething like: `"betterjinja.default": "jinja-html"` or `"betterjinja.j2": "jinja-html"`.

If this results in incorrect file detection (as it is for my case, with linux config files), then one can override this to `"betterjinja.default": "jinja-properties"` or even `"betterjinja.default": ""`.

**Comment by alextremblay** [comment]
I also think that the best course of action would be a configuration option to specify the default jinja "language", that way I could set my personal default for `*.j2` to `jinja-raw`, and @lonix1 could set it to `jinja-properties`, but i don't know how much work that would entail.

I'm only just starting to learn how to write / modify VSCode extensions, so i don't know if I'd be able to contribute to this

**Comment by maciejmatczak** [comment]
Currently the behavior I observe for any `lang` that is not supported, is that my association like:

```text
  "files.associations": {
    ".lang.j2": "jinja"
  },
```

... looses a priority with Better Jinja, where I would like to hook up a _Jira Raw_.

Each time I open a file, same session, it opens with Jira HTML. I can close it and reopen after I changed mode - Better Jira knows better I should use Jira HTML :)
